components/RegisterOffering.js

```diff
@@ -0,0 +1,76 @@
+const noflo = require('noflo');
+const { provider: BigIotProvider } = require('bigiot-js');
+
+exports.getComponent = function () {
+  const c = new noflo.Component();
+  c.description = 'Register offerings with the BIG IoT marketplace';
+  c.icon = 'shopping-cart';
+  c.inPorts.add('offering', {
+    datatype: 'object',
+    description: 'BIG IoT offering definition',
+  });
+  c.inPorts.add('config', {
+    datatype: 'string',
+    description: 'BIG IoT provider config',
+    control: true,
+    scoped: false,
+  });
+  c.inPorts.add('activate', {
+    datatype: 'bang',
+    description: 'Re-activate all offerings',
+  });
+  c.outPorts.add('out', {
+    datatype: 'object',
+  });
+  c.outPorts.add('error', {
+    datatype: 'object',
+  });
+  c.offerings = {};
+  c.tearDown = (callback) => {
+    c.offerings = {};
+    callback();
+  };
+  c.process((input, output) => {
+    if (!input.hasData('config')) {
+      return;
+    }
+
+    if (input.hasData('offering')) {
+      if (!c.offerings[input.scope]) {
+        c.offerings[input.scope] = [];
+      }
+
+      const offering = input.getData('offering');
```

Comment:
Why this special IP reading instead of just checking there is data on all ports?

Reply:
We need to re-register offerings periodically (done in with a RunInterval loop in the Provider graph). So this component executes in two stages:

1. Receive offering(s) and store to component state
2. Receive bang to register all offerings

We do the first regitration when the server has started, and then every 5 minutes. This way we don't register offerings if the server fails to start (for example, invalid port number, or invalid SSL config).